### DISCUSSION TITLE
If you're looking to control your Litra from the command line, I now recommend my new `litra` CLI

### DISCUSSION THREAD

**Discussion by timrogers** [timrogers/litra#313] [Announcements]
Hey all! 👋🏻 

I released the [first version](https://www.npmjs.com/package/litra-glow/v/1.0.0) of `litra` back in September 2022 (as `litra-glow`!) 💡 

Since then, I've shipped many new versions and lots of exciting new features with the help of fantastic [open source contributors](https://github.com/timrogers/litra/graphs/contributors) 🥳 

Most users are looking for a command line tool to control their Litra device(s).

One recurring issue for users trying to do that has been Node.js and npm. Not everyone uses Node.js day-to-day, and setting it up can be tricky - see [this issue](https://github.com/raycast/extensions/issues/6620) for an exmaple.

__To make life easier, I've written a [new version of this tool](https://github.com/timrogers/litra-rs) in [Rust](https://www.rust-lang.org/).__ With Rust, we get the magic of a single, compiled binary which runs anywhere, without dependencies like Node.js or npm.

This new project has all of the features of this one, although the [command line interface](https://github.com/timrogers/litra-rs?tab=readme-ov-file#usage) does look a little different,

**If you're looking for a command line tool to control your Litra device, I now recommend using the `litra` command from the new `timrogers/litra-rs` project**. You can find installation instructions [here](https://github.com/timrogers/litra-rs?tab=readme-ov-file#installation) - and it's especially easy on macOS, since you can install with Homebrew.

I'd only recommend continuing to use this package if you want to integrate Litra controls into a JavaScript application.

If you have any questions or feedback, feel free to leave a comment ❤️

**Comment by jonathonkindle** [comment]
Hey Tim, I came across this project after ALOT of research trying to do exactly what you have done. I want to control the Litra beams via a Stream Deck Plus but am getting hung up on the installation process. After adding the .exe location to path and then trying to run in via "start litra.exe" in command line, I am prompted that the version I downloaded is 16 bit and doesn't work with the 64 bit windows install. Any advice or help you could provide would be amazing.

**Reply by timrogers** [reply]
Hi Jonathon! That sounds quite strange. Can you share a screenshot on the error? I haven't tried Windows myself yet (I will after your message!).

**Reply by timrogers** [reply]
That turned out to be surprisingly painful, but I think I figured it out 😰 Can you try v0.2.1 which I just released?

**Reply by jonathonkindle** [reply]
You... Are.. AMAZING!!!1

Works perfectly now. Working on setting up a command in stream deck to work with these commands.

Thank you so much for the quick response! This really needs to be out there and talked about more than it is, especially with how much hype these lights are getting on youtube.

**Reply by jonathonkindle** [reply]
If you are at all interested in taking a look, you might want to develop an interface for the Streamdeck plugin for Logitech devices like these lights and their cameras. https://marketplace.elgato.com/stream-deck/plugins

Again, thank you for the quick response.

**Comment by jonathonkindle** [comment]
I know that I could probably just switch over to the NPM Javascript version of this project, but I was wondering if there was any way to get the "get" functionality in your rust version?

Stepping up or down the brightness and temperature on the streamdeck requires knowing what the current values are and incrementing up or down. I know your code has some reference to this ability but I know nothing about Rust.

**Comment by timrogers** [comment]
Message ID: < ***@***.*** > Sorry for the slow reply! I've been on vacation 🥳

It's definitely possible to "get" with the new Rust version too. The `devices` command will return current brightness and temperature values, and you can even call with `--json` if you want to use the output from another program.

**Comment by lukehook** [comment]
@jonathonkindle apologies for the late comment on this. I too have just been exploring trying to control my Litra lights via the Stream Deck. It sounds like you made some promising progress here. Did you manage to get things working in the end? If so would you mind sharing any pointers in to how you achieved this? I might see, with permission from maintainers if I can develop an Elgato Marketplace plugin for this

**Comment by jonathonkindle** [comment]
I actually came across a couple of posts exploring the use of the old
version through the stream deck using a batch file. I have been a bit busy
with work but am going to see if I can get it to work with the stream deck
plus knobs in the next week or two. I will attach the url for those posts
when I get back to my computer as well so you can read through it.
